shell/platform/tizen/channels/platform_channel.cc

```diff
@@ -43,7 +48,32 @@ void PlatformChannel::HandleMethodCall(
   } else if (method == "Clipboard.hasStrings") {
     result->NotImplemented();
   } else if (method == "SystemChrome.setPreferredOrientations") {
-    result->NotImplemented();
+    const std::string kPortraitUp = "DeviceOrientation.portraitUp";
+    const std::string kPortraitDown = "DeviceOrientation.portraitDown";
+    const std::string kLandscapeLeft = "DeviceOrientation.landscapeLeft";
+    const std::string kLandscapeRight = "DeviceOrientation.landscapeRight";
+    const std::map<std::string, int> orientation_mapping = {
+        {kPortraitUp, 0},
+        {kLandscapeLeft, 90},
+        {kPortraitDown, 180},
+        {kLandscapeRight, 270},
+    };
+
+    const auto& list = call.arguments()[0];
+    std::vector<int> rotations;
+    for (rapidjson::Value::ConstValueIterator itr = list.Begin();
```

Comment:
Consider logging values passed by the user with `FT_LOGD` to make debugging easier.